# This is a patch for the Rudder server cfengine paths
# (independent from the Linux flavour on which the Rudder server itself is running)
#
# This patch allows for a better handling of basic Linux utilities paths 
# for nodes running ArchLinux or Manjaro Linux.
#
# Without this patch, Rudder techniques using calls to utilities such as "usermod"
# may fail in error.
#
# Applying this patch on a Rudder server will not affect nodes running any non-ArchLiux
# family Linuxes.
#
# Apply this patch on an already installed Rudder server by :
#
# patch /opt/rudder/share/initial-ncf/20_cfe_basics/cfengine/paths.cf < 9900_rudder-server_postinstall_cfengine_paths.patch
# patch /usr/share/ncf/tree/20_cfe_basics/cfengine/paths.cf < 9900_rudder-server_postinstall_cfengine_paths.patch
#
# Then :
# rudder server trigger-policy-generation
#
# The patch will need to be reapplied if the Rudder server is reinstalled or upgraded

diff -aurN rudder-sources-8.3.1~git202505042104/ncf/tree/20_cfe_basics/cfengine/paths.cf rudder-sources.patched/ncf/tree/20_cfe_basics/cfengine/paths.cf
--- rudder-sources-8.3.1~git202505042104/ncf/tree/20_cfe_basics/cfengine/paths.cf	2025-05-04 21:04:52.000000000 +0200
+++ rudder-sources.patched/ncf/tree/20_cfe_basics/cfengine/paths.cf	2025-05-05 06:59:20.841311758 +0200
@@ -214,9 +214,14 @@
       "path[tr]"                string => "/usr/bin/tr";
       #
       "path[pacman]"            string => "/usr/bin/pacman";
+      "path[pamac]"             string => "/usr/bin/pamac";
       "path[yaourt]"            string => "/usr/bin/yaourt";
       "path[useradd]"           string => "/usr/bin/useradd";
+      "path[userdel]"           string => "/usr/bin/userdel";
+      "path[usermod]"           string => "/usr/bin/usermod";
       "path[groupadd]"          string => "/usr/bin/groupadd";
+      "path[groupdel]"          string => "/usr/bin/groupdel";
+      "path[groupmod]"          string => "/usr/bin/groupmod";
       "path[ip]"                string => "/usr/bin/ip";
       "path[ifconfig]"          string => "/usr/bin/ifconfig";
       "path[journalctl]"        string => "/usr/bin/journalctl";
